microsetta_admin/templates/per_sample_summary.html

```diff
@@ -27,36 +42,50 @@
 {% block content %}
 <h3>Sample Summaries</h3>
 <hr class="dashed">
-<h5>Retrieve summary for a single sample</h5>
+{% if error_message %}
+<p style="color:red">
+    {{ error_message |e }}
+</p>
+{% endif %}
 <div style="height: 400px; width: 400px">
-    {% if error_message %}
-        <p style="color:red">
-            {{ error_message |e }}
-        </p>
-    {% endif %}
-    <form name="search_form" id="search_form" method="GET">
-        <table width=100% border=0>
-        <tr>
-            <td colspan=3 style="text-align:left;padding:4px"> <input type="text" name="sample_barcode" placeholder="Enter barcode here" id="sample_barcode" {% if info %} value="{{info.barcode}}"{% endif %}> </td>
-            <td colspan=1 style="text-align:right;padding:4px"> <input type="submit" value="Retrieve"> </td>
-        </tr>
-        </table>
-        <script> document.search_form.sample_barcode.focus() </script>
-    </form>
-    <hr class="dashed">
-    <h5>Retrieve summaries using a csv file</h5>
-    <form name="upload_csv" id="search_csv" method="POST" enctype="multipart/form-data" onsubmit="return remove_error_messages()">
+    <h5>Retrieve summary for a sample</h5>
+    <form name="search_form" id="search_form" method="POST" enctype="multipart/form-data" onsubmit="return handleFormSubmit(event)">
         <table width=100% border=0>
             <tr>
-                <td colspan=3 style="text-align:left;padding:4px"><input type=file name=file></td>
-                <td colspan=1 style="text-align:right;padding:4px"><input type=submit value=Upload></td>
+                <td colspan=3 style="text-align:left;padding:4px">
+                    <label for="search_field">Select search field:</label>
+                    <select id="search_field" name="search_field">
+                        <option value="sample_barcode">Barcode</option>
+                        <option value="kit_id">Kit ID</option>
+                        <option value="email">Email</option>
+                        <option value="outbound_tracking">Outbound Tracking Number</option>
+                        <option value="inbound_tracking">Inbound Tracking Number</option>
+                    </select>
+                </td>
+            </tr>
+            <tr>
+                <td colspan=3 style="text-align:left;padding:4px">
+                    <input type="text" name="text_input" placeholder="Enter value here" id="text_input">
+                </td>
+            </tr>
+            <tr>
+                <td colspan=3 style="text-align:left;padding:4px">
+                    <input type="file" name="file" id="file">
+                </td>
+            </tr>
```

Comment:
Can you please clearly label the fields, e.g. Single Search for the text input and Upload List for the file input? Also, can you either prevent both from being submitted or indicate on the UI which takes precedence if both have values?

Reply:
@ayobi The code to prevent both from being submitted looks good, but can you please clearly label the fields? Thanks.

Reply:
Gotcha. I updated the vars to reflect the id `upload_list` and `single_search`.